ocs_ci/framework/pytest_customization/ocscilib.py

```diff
@@ -362,17 +362,19 @@ def pytest_configure(config):
                     "Skipping version collection because we skipped "
                     "the OCS deployment"
                 )
-                return
+                continue
             elif ocsci_config.RUN["cli_params"].get("dev_mode"):
                 log.info("Running in development mode")
-                return
+                continue
             # remove extraneous metadata
             for extra_meta in ["Python", "Packages", "Plugins", "Platform"]:
                 if config._metadata.get(extra_meta):
                     del config._metadata[extra_meta]
 
             config._metadata["Test Run Name"] = get_testrun_name()
             gather_version_info_for_report(config)
+    # switch the configuration context back to the first cluster
+    ocsci_config.switch_ctx(0)
```

Comment:
Shouldn't we switch back to the default cluster context passed via --default-cluster-context-index ?  Should be accessible via config.ENV_DATA["default_cluster_context_index"] I think.

Reply:
Good catch! Fixing in [PR#5413](https://github.com/red-hat-storage/ocs-ci/pull/5413).